File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::map<int, bool> seen{};

        for (auto num : nums) {
            if (seen[num]) 
                return true;

            seen[num] = true;            
        }

        return false;
    }
};

File: Data Structures & Algorithms/majority-element/submission-0.py
from collections import defaultdict

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        seen = defaultdict(int)

        for num in nums:
            seen[num] += 1

        for num, count in seen.items():
            if count > (len(nums) // 2):
                return num